components/base/kinematicbase/ptgKinematics.go

```diff
@@ -218,20 +221,13 @@ func (ptgk *ptgBaseKinematics) ErrorState(ctx context.Context, plan [][]referenc
 	}
 	actualPIF := spatialmath.PoseBetween(ptgk.origin, actualPIFRaw.Pose())
 
-	var nominalPose spatialmath.Pose
-
 	// Determine the nominal pose, that is, the pose where the robot ought be if it had followed the plan perfectly up until this point.
-	// This is done differently depending on what sort of frame we are working with.
-	// TODO: The `rectifyTPspacePath` in motionplan does basically this. Deduplicate.
-	runningPose := spatialmath.NewZeroPose()
-	for i := 0; i < currentNode; i++ {
-		wp := plan[i]
-		wpPose, err := ptgk.frame.Transform(wp)
-		if err != nil {
-			return nil, err
-		}
-		runningPose = spatialmath.Compose(runningPose, wpPose)
+	// TODO: double check this math
+	poses, err := path.GetFramePoses(ptgk.Name().ShortName())
+	if err != nil {
+		return nil, err
 	}
+	nominalPose := poses[waypointIndex-1]
```

Comment:
question for my own understanding.
does `poses[waypointIndex]` gives  our current pose?
then `poses[waypointIndex -1]` gives us our former running pose?

Reply:
yes